File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    
    phone_book.sort()
    
    for i in range(len(phone_book)-1) :
        
        save = phone_book[i]
        comp = phone_book[i+1]
        
        if save == comp[:len(save)] :
            return False

    
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        Arrays.sort(phone_book);
        
        for (int i= 0; i < phone_book.length-1; i++)
        {
            int strsize = phone_book[i].length();
            if(phone_book[i+1].length() > strsize)
            {
                if(phone_book[i+1].substring(0,strsize).equals(phone_book[i]))
                {
                    answer=false;
                    break;
                }
            }
        }
        
        return answer;
    }
}